perun-ldapc/src/main/java/cz/metacentrum/perun/ldapc/processor/impl/FacilityAttributeProcessor.java

```diff
@@ -37,12 +39,23 @@ public void processAttributeSet(String msg, MessageBeans beans) {
 			log.debug("Getting resources assigned to facility {}", beans.getFacility().getId());
 			// List<Resource> resources = Rpc.FacilitiesManager.getAssignedResources(ldapcManager.getRpcCaller(), beans.getFacility());
 			List<Resource> resources = perun.getFacilitiesManager().getAssignedResources(ldapcManager.getPerunSession(), beans.getFacility());
+
+			Attribute virtAttributeWithValue = null;
+			if (beans.getAttribute().getNamespace().contains(":virt")) {
+				virtAttributeWithValue = ldapcManager.getPerunBl().getAttributesManager().getAttribute(ldapcManager.getPerunSession(),
+					beans.getFacility(), beans.getAttribute().getName());
+			}
+
 			for(Resource resource: resources) {
 				log.debug("Setting attribute {} for resource {}", beans.getAttribute(), resource);
-				perunResource.modifyEntry(resource, beans.getAttribute());
+				if (virtAttributeWithValue != null) {
+					perunResource.modifyEntry(resource, virtAttributeWithValue);
+				} else {
+					perunResource.modifyEntry(resource, beans.getAttribute());
+				}
 			}
-		} catch (NamingException | FacilityNotExistsException | InternalErrorException | PrivilegeException e) {
-				log.error("Error setting attribute:", e);
+		} catch (PerunException e) {
```

Comment:
You shouldn't hide all exception behind single PerunException. Also NamingException was previously catched and now it would be thrown outside probably stopping LDAPc process.